src/core/sequential_write_dataframe.jl

```diff
@@ -0,0 +1,52 @@
+struct SequentialWriteDataFrame <: DataFrames.AbstractDataFrame
+    data::DataFrames.DataFrame
+    last_recorded_row::Base.RefValue{Int}
+    SequentialWriteDataFrame(args...; kw...) =
+        new(DataFrames.DataFrame(args...; kw...), Ref(0))
+    SequentialWriteDataFrame(df::DataFrames.DataFrame) = new(df, Ref(0))
+end
+
+get_last_recorded_row(df::SequentialWriteDataFrame) = df.last_recorded_row[]
+
+function set_last_recorded_row!(df::SequentialWriteDataFrame, val::Int)
```

Comment:
Shouldn't this get called by one of the set functions?

Reply:
working on it... the implementation of `setindex!()` isn't obvious here